assimilation_code/location/threed_cartesian/location_mod.f90

```diff
@@ -67,32 +68,56 @@ module location_mod
 ! This version supports only regularly spaced boxes (the non-working octree code
 ! was removed.) 
 
+!type box_type
+!   private
+!   integer, pointer  :: loc_box(:)           ! (nloc); List of loc indices in boxes
+!   integer, pointer  :: count(:, :, :)       ! (nx, ny, nz); # of locs in each box
+!   integer, pointer  :: start(:, :, :)       ! (nx, ny, nz); Start of list of locs in this box
+!   real(r8)          :: bot_x, top_x         ! extents in x, y, z
+!   real(r8)          :: bot_y, top_y
+!   real(r8)          :: bot_z, top_z
+!   real(r8)          :: x_width, y_width, z_width    ! widths of boxes in x,y,z
+!   real(r8)          :: nboxes_x, nboxes_y, nboxes_z ! based on maxdist how far to search
+!end type box_type
+
 type box_type
    private
-   integer, pointer  :: loc_box(:)           ! (nloc); List of loc indices in boxes
-   integer, pointer  :: count(:, :, :)       ! (nx, ny, nz); # of locs in each box
-   integer, pointer  :: start(:, :, :)       ! (nx, ny, nz); Start of list of locs in this box
+   integer           :: num                      
+   real(r8)          :: maxdist              ! furthest seperation between "close" locations  
+   integer, allocatable  :: loc_box(:)           ! (nloc); List of loc indices in boxes
+   integer, allocatable  :: count(:, :, :)       ! (nx, ny, nz); # of locs in each box
+   integer, allocatable  :: start(:, :, :)       ! (nx, ny, nz); Start of list of locs in this box
    real(r8)          :: bot_x, top_x         ! extents in x, y, z
    real(r8)          :: bot_y, top_y
    real(r8)          :: bot_z, top_z
    real(r8)          :: x_width, y_width, z_width    ! widths of boxes in x,y,z
    real(r8)          :: nboxes_x, nboxes_y, nboxes_z ! based on maxdist how far to search
 end type box_type
 
-! Type to facilitate efficient computation of observations close to a given location
+
+
 type get_close_type
    private
-   integer           :: num
-   real(r8)          :: maxdist
-   type(box_type)    :: box
+   integer                     :: nt                      ! The number of distinct cutoffs
+   !real(r8),allocatable        :: type_to_cutoff_map(:)   ! mapping of types to index
+   integer, allocatable        :: type_to_cutoff_map(:)   ! mapping of types to index
+   type(box_type),allocatable  :: box(:)                  ! Array of box types 
 end type get_close_type
 
+
+!type get_close_type
+!   private
+!   integer           :: num
+!   real(r8)          :: maxdist
+!   type(box_type)    :: box
+!end type get_close_type
+
 type(random_seq_type) :: ran_seq
 logical               :: ran_seq_init = .false.
 logical, save         :: module_initialized = .false.
 
 character(len = 512) :: errstring
-
+character(len = 512) :: msgstring, msgstring1, msgstring2
 
```

Comment:
sorry - i can't figure out how to leave a comment in the unchanged part of the code.  the comment block that starts below on lines 154 - 157  should be removed, right?  because that's what this pull request is adding - per-type localization distances as an option.